daemon/src/projection.rs

```diff
@@ -211,12 +228,51 @@ struct Aggregated {
     collab_settlement_tx: Option<(Transaction, Script)>,
     /// If this is present, it should have been published.
     cet: Option<Transaction>,
-    closing_price: Option<Price>,
+
+    /// If this is present the cet has not been published
+    timelocked_cet: Option<Transaction>,
 
     commit_published: bool,
     refund_published: bool,
 }
 
+impl Aggregated {
+    fn payout(self, role: Role) -> Option<Amount> {
+        if let Some((tx, script)) = self.collab_settlement_tx {
+            return Some(extract_payout_amount(tx, script));
+        }
+
+        let tx = self.cet.or(self.timelocked_cet)?;
+        let dlc = self
+            .latest_dlc
+            .as_ref()
+            .expect("dlc to be present when we have a cet");
```

Comment:
You could `?` this instead of panicking.

Reply:
I think this one should really never happen and I wanted to avoid returning a a `Result` from `payout`. But maybe it's better to go with a `Result` then.

Reply:
It is an option so the None will bubble up.